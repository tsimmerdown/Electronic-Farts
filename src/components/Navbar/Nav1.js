import React from "react"
import styled from "styled-components"
import { FaRegUser } from "react-icons/fa"
import { AiOutlineQuestion } from "react-icons/ai"
const Cont = styled.div`
  width: 100%;
  background: rgb(22, 22, 22);
  height: 40px;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  color: rgba(243, 243, 243, 0.7);
`

const Profile = styled(FaRegUser)`
  font-size: 18px;
  margin-right: 30px;

  &:hover {
    color: rgba(243, 243, 243);
  }
`

const Help = styled(AiOutlineQuestion)`
  font-size: 22px;
  margin-right: 30px;
  &:hover {
    color: rgba(243, 243, 243);
  }
`
const Logo = styled.div`
  font-family: EA Logo;
  font-size: 20px;
  margin-right: 45px;
  color: rgba(243, 243, 243, 0.4);
  &:hover {
    color: rgba(243, 243, 243, 0.7);
  }
`

const Nav1 = () => {
  return (
    <Cont>
      <Profile />
      <Help />
      <Logo>EA</Logo>
    </Cont>
  )
}

export default Nav1
