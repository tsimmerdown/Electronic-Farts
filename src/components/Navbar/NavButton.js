import React, { useState } from "react"
import styled from "styled-components"
import { Popover, Divider } from "@material-ui/core"

const Cont = styled.div`
  background: #fff;
  height: 100%;
  display: flex;
  align-items: center;
  margin-left: 34px;
`
const ListText = styled.div`
  height: 30px;
  font-size: 15px;
  color: rgba(22, 22, 22, 0.6);
  &:hover {
    color: red;
  }
`
const Text = styled.div`
  &:hover {
    color: red;
  }
`

const NavButton = ({ title, list }) => {
  const [anchorEl, setAnchorEl] = useState(null)

  const handleClick = event => {
    setAnchorEl(event.currentTarget)
  }

  const handleClose = () => {
    setAnchorEl(null)
  }

  const open = Boolean(anchorEl)

  return (
    <>
      {list ? (
        <Cont onMouseEnter={handleClick} onMouseLeave={handleClose}>
          <Text data-testid="text" style={{ fontSize: "18px" }}>
            {title}
          </Text>
          <Popover
            style={{ pointerEvents: "none" }}
            open={open}
            anchorEl={anchorEl}
            onClose={handleClose}
            anchorOrigin={{
              vertical: "bottom",
              horizontal: "left",
            }}
            transformOrigin={{
              vertical: "top",
              horizontal: "left",
            }}
            disableRestoreFocus
          >
            <div
              style={{ pointerEvents: "auto", padding: "20px  35px 10px 35px" }}
            >
              {list.map((item, key) => {
                return <ListText key={key}>{item}</ListText>
              })}
            </div>
          </Popover>
        </Cont>
      ) : (
        <div>
          <Divider style={{ margin: "0 24px" }} />
          <Text
            style={{ fontSize: "18px", padding: "18px 0", margin: "0 24px" }}
            data-testid="text"
          >
            {title}
          </Text>
        </div>
      )}
    </>
  )
}

export default NavButton
