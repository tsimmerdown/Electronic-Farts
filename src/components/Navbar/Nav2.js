import React, { useState } from "react"
import { Link } from "gatsby"
import styled from "styled-components"
import { BsThreeDotsVertical } from "react-icons/bs"
import { GiHamburgerMenu } from "react-icons/gi"
import { NavbarData } from "../../data/NavbarData"
import NavButton from "./NavButton"
import { Drawer } from "@material-ui/core"
import { useMediaQuery } from "react-responsive"
import { GrClose } from "react-icons/gr"
import { responsive } from "../../data/responsive"

const Cont = styled.div`
  width: 100%;
  background: #fff;
  height: 56px;
  display: flex;
  align-items: center;
  padding-left: 15px;
`

const Title = styled(Link)`
  color: red;
  text-decoration: none;
  font-size: 27px;
  font-weight: 600;
  margin-left: 15px;
`
const Dots = styled(BsThreeDotsVertical)`
  &:hover {
    transform: scale(1.4);
    color: red;
  }
  transition: all 0.3s ease-in-out;
`

const BurgerCont = styled.div`
  position: absolute;
  right: 0;
  transform: translate(-100%, 0);
  display: ${props => (props.isBigScreen ? "none" : "block")};
`

const BurgerIcon = styled(GiHamburgerMenu)`
  font-size: 25px;
  &:hover {
    color: red;
  }
`
const DrawerCont = styled(Drawer)`
  && {
    top: 0;
    left: 0;
    height: 100vh;
    width: 100vw;
  }
`
const Header = styled.div`
  margin: 20px 0 0 20px;
  height: 56px;
`
const Close = styled(GrClose)`
  position: absolute;
  right: 0;
  transform: translate(-20px, 50%);
  &:hover {
    color: red;
  }
`
const Nav2 = () => {
  const [openDrawer, setOpenDrawer] = useState(false)
  const isBigScreen = useMediaQuery({
    query: `(min-width: ${responsive.tablet}px)`,
  })

  const handleDrawerOpen = () => {
    setOpenDrawer(!openDrawer)
  }

  return (
    <Cont>
      <Dots
        style={{ fontSize: "24px" }}
        whileHover={{
          scale: 1.1,
        }}
        onClick={() => {
          console.log(isBigScreen)
        }}
      />
      <Title to="/" data-testid="title">
        Electronic Farts
      </Title>
      {!isBigScreen ? (
        <>
          <BurgerCont isBigScreen={isBigScreen}>
            <BurgerIcon onClick={handleDrawerOpen} />
          </BurgerCont>
          <DrawerCont anchor="top" open={openDrawer}>
            <Header>
              <Title to="/" data-testid="title">
                Electronic Farts
              </Title>
              <Close onClick={handleDrawerOpen} />
            </Header>
            <div style={{ padding: "21px 0 56px 0" }}>
              {NavbarData.map((item, key) => {
                return (
                  <NavButton
                    data-testid="button"
                    title={item.title}
                    key={key}
                  />
                )
              })}
            </div>
          </DrawerCont>
        </>
      ) : (
        NavbarData.map((item, key) => {
          return <NavButton title={item.title} list={item.sub} key={key} />
        })
      )}
    </Cont>
  )
}

export default Nav2
