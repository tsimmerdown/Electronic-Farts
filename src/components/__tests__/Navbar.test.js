import React from "react"
import { render } from "@testing-library/react"
import renderer from "react-test-renderer"
import Nav1 from "../Navbar/Nav1"
import Nav2 from "../Navbar/Nav2"
import NavButton from "../Navbar/NavButton"

describe("Navbar Component", () => {
  it("matches snapshot Nav1", () => {
    const tree = renderer.create(<Nav1 />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("renders without crashing", () => {
    const { getByTestId } = render(<Nav2 />)
    expect(getByTestId("title")).toHaveTextContent("Electronic Farts")
  })

  it("matches snapshot Nav2", () => {
    const tree = renderer.create(<Nav2 />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("matches snapshot NavButton", () => {
    const tree = renderer.create(<NavButton />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly Navbutton", () => {
    const { getByTestId } = render(<NavButton title="Game" />)
    expect(getByTestId("text")).toHaveTextContent("Game")
  })
})
