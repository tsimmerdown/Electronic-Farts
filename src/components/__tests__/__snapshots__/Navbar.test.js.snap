// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component matches snapshot Nav1 1`] = `
<div
  className="sc-bdvvaa jbUJQp"
>
  <svg
    className="sc-gsDJrp dwoyio"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M313.6 304c-28.7 0-42.5 16-89.6 16-47.1 0-60.8-16-89.6-16C60.2 304 0 364.2 0 438.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-25.6c0-74.2-60.2-134.4-134.4-134.4zM400 464H48v-25.6c0-47.6 38.8-86.4 86.4-86.4 14.6 0 38.3 16 89.6 16 51.7 0 74.9-16 89.6-16 47.6 0 86.4 38.8 86.4 86.4V464zM224 288c79.5 0 144-64.5 144-144S303.5 0 224 0 80 64.5 80 144s64.5 144 144 144zm0-240c52.9 0 96 43.1 96 96s-43.1 96-96 96-96-43.1-96-96 43.1-96 96-96z"
    />
  </svg>
  <svg
    className="sc-dkPtyc iizalO"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 1024 1024"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M764 280.9c-14-30.6-33.9-58.1-59.3-81.6C653.1 151.4 584.6 125 512 125s-141.1 26.4-192.7 74.2c-25.4 23.6-45.3 51-59.3 81.7-14.6 32-22 65.9-22 100.9v27c0 6.2 5 11.2 11.2 11.2h54c6.2 0 11.2-5 11.2-11.2v-27c0-99.5 88.6-180.4 197.6-180.4s197.6 80.9 197.6 180.4c0 40.8-14.5 79.2-42 111.2-27.2 31.7-65.6 54.4-108.1 64-24.3 5.5-46.2 19.2-61.7 38.8a110.85 110.85 0 0 0-23.9 68.6v31.4c0 6.2 5 11.2 11.2 11.2h54c6.2 0 11.2-5 11.2-11.2v-31.4c0-15.7 10.9-29.5 26-32.9 58.4-13.2 111.4-44.7 149.3-88.7 19.1-22.3 34-47.1 44.3-74 10.7-27.9 16.1-57.2 16.1-87 0-35-7.4-69-22-100.9zM512 787c-30.9 0-56 25.1-56 56s25.1 56 56 56 56-25.1 56-56-25.1-56-56-56z"
    />
  </svg>
  <div
    className="sc-hKwCoD iJQPmY"
  >
    EA
  </div>
</div>
`;

exports[`Navbar Component matches snapshot Nav2 1`] = `
<div
  className="sc-iCfLBT dKBZdA"
>
  <svg
    className="sc-pVTma dKjdAS"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
        "fontSize": "24px",
      }
    }
    viewBox="0 0 16 16"
    whileHover={
      Object {
        "scale": 1.1,
      }
    }
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M9.5 13a1.5 1.5 0 11-3 0 1.5 1.5 0 013 0zm0-5a1.5 1.5 0 11-3 0 1.5 1.5 0 013 0zm0-5a1.5 1.5 0 11-3 0 1.5 1.5 0 013 0z"
      fillRule="evenodd"
    />
  </svg>
  <a
    className="sc-furvIG PyjFS"
    data-testid="title"
    href="/"
  >
    Electronic Farts
  </a>
  <div
    className="sc-jrQzUz DvJSX"
  >
    <svg
      className="sc-kDThTU ifMCWc"
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
      />
    </svg>
  </div>
</div>
`;

exports[`Navbar Component matches snapshot NavButton 1`] = `
<div>
  <hr
    className="MuiDivider-root"
    style={
      Object {
        "margin": "0 24px",
      }
    }
  />
  <div
    className="sc-gKckTs kjfbny"
    data-testid="text"
    style={
      Object {
        "fontSize": "18px",
        "margin": "0 24px",
        "padding": "18px 0",
      }
    }
  />
</div>
`;
